percy/metadata/metadata.js

```diff
@@ -16,12 +16,20 @@ class Metadata {
     this._orientation = orientation || 'caps';
     this._statusBarHeight = statusBarHeight;
     this._navigationBarHeight = navigationBarHeight;
+    this._viewportRect = null;
   }
 
   // items that need caps are moved to getters as caps are not stored on wd driver object
   // So we need to make a lazy call to avoid making session get call in app automate context
   async caps() {
-    return await this.driver.getCapabilities();
+    /* istanbul ignore next */
+    if (typeof browser !== 'undefined') {
```

Comment:
why are we accessing global browser here ? it shouldnt be directly accessed, we already have reference in this.driver

Reply:
updated to use driver.wdio

Reply:
No i mean we should do any driver specific things in the adaptor class [ the driverWrapper class ] and not in any other part of codebase.  you can update methods there

Reply:
Also while this seems to rely on global browser object to be available, which is usually true in wdio, but its not necessary. Users can also write a script that does not use global browser [ and so adaptor checks for type of passed driver instead ]

Reply:
@ninadbstack yes i have moved caps in driverWrapper only, others which were specific to ios only I have kept in iosMetadata.